services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/ResumableFileDownload.java

```diff
@@ -125,6 +152,153 @@ public Optional<Long> totalSizeInBytes() {
         return Optional.ofNullable(totalSizeInBytes);
     }
 
+    @Override
+    public String toString() {
+        return ToString.builder("ResumableFileDownload")
+                       .add("bytesTransferred", bytesTransferred)
+                       .add("fileLastModified", fileLastModified)
+                       .add("s3ObjectLastModified", s3ObjectLastModified)
+                       .add("totalSizeInBytes", totalSizeInBytes)
+                       .add("downloadFileRequest", downloadFileRequest)
+                       .build();
+    }
+
+    /**
+     * Persists this download object to a file in Base64-encoded JSON format.
+     *
+     * @param path The path to the file to which you want to write the serialized download object.
+     */
+    public void writeToFile(Path path) {
+        try {
+            Files.write(path, toBase64Bytes(ResumableFileDownloadSerializer.toJson(this)));
```

Comment:
It seems we are serializing every time this method is invoked. Same for other ser/deser methods. I think it might be fine for now and we can optimize later if customers report issue. WDYT?

Reply:
Yes. My reasoning is that since a `ResumableFileDownload` is a transient type of object, storing the serialized version wouldn't improve performance a lot. I expect the pattern would be pause download -> serialize to the format you want once -> garbage collect.